src/librustc_ty/instance.rs

```diff
@@ -119,17 +114,10 @@ fn resolve_associated_item<'tcx>(
                 trait_closure_kind,
             ))
         }
-        traits::VtableFnPointer(ref data) => {
-            // `FnPtrShim` requires a monomorphic aka concrete type.
-            if data.fn_ty.needs_subst() {
-                return None;
-            }
-
-            Some(Instance {
-                def: ty::InstanceDef::FnPtrShim(trait_item.def_id, data.fn_ty),
-                substs: rcvr_substs,
-            })
-        }
+        traits::VtableFnPointer(ref data) => Some(Instance {
```

Comment:
This needs to check that the type is a `ty::FnPtr`. I guess you could assert it, since `VtableFnPointer` wouldn't be created if it weren't.

Reply:
Fixed in 2401187.